### PR TITLE
[quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39826 [quant][graphmode] Quantizing traced modules
* #39825 [quant][graphmode] Remove dedup pass
* **#39645 [quant][graphmode] Support quantized::batch_norm2d_relu fusion for tracing**

Summary:
This PR added quantization support for handling BatchNorm2d and ReLU(or F.relu) in both
scripting and tracing

Test Plan:
python test/test_quantization.py TestQuantizeScriptPTSQOps.test_qbatchnorm_relu
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21942111](https://our.internmc.facebook.com/intern/diff/D21942111)

### CODE REVIEW

File: test/quantization/test_quantize_script.py [hunk 1]
```diff
@@ -1868,11 +1867,12 @@ def forward(self, x):
 
         data = [(torch.rand((1, 3, 10, 10), dtype=torch.float), torch.randint(0, 1, (1,), dtype=torch.long)) for _ in range(2)]
         for instance in [BNRelu(True), BNRelu(False), BNFuncRelu(), BNFuncInplaceRelu()]:
-            model = self._test_op_impl(instance, data, "quantized::batch_norm2d_relu")
-            FileCheck().check_not("aten::batch_norm") \
-                       .check_not("aten::relu") \
-                       .check_not("aten::relu_") \
-                       .run(model.graph)
+            for tracing in [True, False]:
```
Comment (raghuramank100) [inline test/quantization/test_quantize_script.py:1870]: This test: test_qbatch_norm_relu can only run in FBGEMM as batch norm is not supported as an op in qnnpack, please add a check for qengine for this test.

**Review by raghuramank100 — APPROVED** [review]

File: test/quantization/test_quantize_script.py (hunk 1 above)
Comment (jerryzh168) [reply test/quantization/test_quantize_script.py:1870]: OK, I'll do skipIfNoFBGEMM
